app/src/deploys/deploys.service.ts

```diff
@@ -1,16 +1,46 @@
-import { Injectable } from "@nestjs/common";
+import { CACHE_MANAGER, Inject, Injectable } from "@nestjs/common";
+import { Cache } from "cache-manager";
 import { CLValueParsers } from "casper-js-sdk";
+import { StatusCodes } from "http-status-codes";
 import { onChain } from "src/main";
+import { SidecarDeploy } from "src/types/api";
 import { DeployStatus, GetDeploy } from "src/types/deploy";
+import { ApiError } from "src/utils/ApiError";
 import {
   determineDeploySessionData,
   JsonDeploySession,
 } from "src/utils/deploy";
 
 @Injectable()
 export class DeploysService {
+  constructor(@Inject(CACHE_MANAGER) private readonly cacheManager: Cache) {}
+
   async getDeploy(hash: string): Promise<GetDeploy> {
-    const { deploy, executionResults } = await onChain.getDeploy(hash);
+    const cachedDeployByHash = await this.cacheManager.get<SidecarDeploy>(hash);
+
+    let deploy: SidecarDeploy["deploy_accepted"];
+    let executionResults: {
+      block_hash: string;
+      // TODO: properly type this as part of ticket #94
+      /* eslint-disable  @typescript-eslint/no-explicit-any */
+      result: any;
+    }[];
+    if (cachedDeployByHash) {
+      executionResults = [
+        {
+          block_hash: cachedDeployByHash.deploy_processed.block_hash,
+          result: cachedDeployByHash.deploy_processed.execution_result,
+        },
+      ];
+
+      deploy = cachedDeployByHash.deploy_accepted;
+    } else {
+      // TODO: properly type this as part of ticket #94
+      const processedDeploy = await onChain.getDeploy(hash);
```

Comment:
Same question here

Reply:
If you look a little deeper into the `onChain` method you will see that there is indeed a try/catch wrapping the sidecar/rpc methods. This function is just a wrapper for those.

Reply:
Ahh okay sweet thanks for clarifying dude!

Reply:
Np!